clients/unshaded/src/test/java/tachyon/client/file/AbstractTachyonFileSystemTest.java

```diff
@@ -0,0 +1,366 @@
+    Mockito.verify(mFileSystemMasterClient).delete(FILE_ID, true);
+  }
+
+  @Test
+  public void deleteExceptionTest() throws Exception {
+    DeleteOptions deleteOptions = new DeleteOptions.Builder().setRecursive(true).build();
+    Mockito.when(mFileSystemMasterClient.delete(FILE_ID, true)).thenThrow(EXCEPTION);
+    try {
+      mFileSystem.delete(new TachyonFile(FILE_ID), deleteOptions);
+      Assert.fail("Exception should have been propagated");
+    } catch (Exception e) {
+      Assert.assertSame(EXCEPTION, e);
+    }
+  }
+
+  @Test
+  public void freeTest() throws Exception {
+    FreeOptions freeOptions = new FreeOptions.Builder().setRecursive(true).build();
+    mFileSystem.free(new TachyonFile(FILE_ID), freeOptions);
+    Mockito.verify(mFileSystemMasterClient).free(FILE_ID, true);
+  }
+
+  @Test
+  public void freeExceptionTest() throws Exception {
+    FreeOptions freeOptions = new FreeOptions.Builder().setRecursive(true).build();
+    Mockito.when(mFileSystemMasterClient.free(FILE_ID, true)).thenThrow(EXCEPTION);
+    try {
+      mFileSystem.free(new TachyonFile(FILE_ID), freeOptions);
+      Assert.fail("Exception should have been propagated");
+    } catch (Exception e) {
+      Assert.assertSame(EXCEPTION, e);
+    }
+  }
+
+  @Test
+  public void getInfoTest() throws Exception {
+    FileInfo info = new FileInfo();
+    Mockito.when(mFileSystemMasterClient.getFileInfo(FILE_ID)).thenReturn(info);
+    GetInfoOptions getInfoOptions = GetInfoOptions.defaults();
+    Assert.assertSame(info, mFileSystem.getInfo(new TachyonFile(FILE_ID), getInfoOptions));
+    Mockito.verify(mFileSystemMasterClient).getFileInfo(FILE_ID);
+  }
+
+  @Test
+  public void getInfoExceptionTest() throws Exception {
+    Mockito.when(mFileSystemMasterClient.getFileInfo(FILE_ID)).thenThrow(EXCEPTION);
+    GetInfoOptions getInfoOptions = GetInfoOptions.defaults();
+    try {
+      mFileSystem.getInfo(new TachyonFile(FILE_ID), getInfoOptions);
+      Assert.fail("Exception should have been propagated");
+    } catch (Exception e) {
+      Assert.assertSame(EXCEPTION, e);
+    }
+  }
+
+  @Test
+  public void listStatusTest() throws Exception {
+    List<FileInfo> infos = Lists.newArrayList(new FileInfo());
+    Mockito.when(mFileSystemMasterClient.getFileInfoList(FILE_ID)).thenReturn(infos);
+    ListStatusOptions listStatusOptions = ListStatusOptions.defaults();
+    Assert.assertSame(infos, mFileSystem.listStatus(new TachyonFile(FILE_ID), listStatusOptions));
+    Mockito.verify(mFileSystemMasterClient).getFileInfoList(FILE_ID);
+  }
+
+  @Test
+  public void listStatusExceptionTest() throws Exception {
+    Mockito.when(mFileSystemMasterClient.getFileInfoList(FILE_ID)).thenThrow(EXCEPTION);
+    ListStatusOptions listStatusOptions = ListStatusOptions.defaults();
+    try {
+      mFileSystem.listStatus(new TachyonFile(FILE_ID), listStatusOptions);
+      Assert.fail("Exception should have been propagated");
```

Comment:
Since it's only for test code I think it's easier to keep it like this - the pros for moving this to a constants class are easier auto-complete and consistency, but these are really minor. The cons are extra work to maintain the consistency, and dealing with extra indirection when reading the test.